src/account.ts

```diff
@@ -36,72 +52,104 @@ export const getSubprotocol = (message: BtpPacket, name: string) =>
   private master: EthereumPlugin
   // Send the given BTP packet message to this counterparty
   private sendMessage: (message: BtpPacket) => Promise<BtpPacketData>
-  // Data handler mapped from plugin
+  // Data handler from plugin for incoming ILP packets
   private dataHandler: DataHandler
-  // Money handler mapped from plugin
+  // Money handler from plugin for incoming money
   private moneyHandler: MoneyHandler
-  // Queue to handle all incoming settlements/BTP transfers
-  private incomingSettlements = new Mutex()
-  // Queue to handle all outgoing settlements
-  // If a settlement is occuring, only a single settlement can be queued
-  private outgoingSettlements = new Mutex(2)
-  // Queue of (at most) a single transaction to claim the incoming channel
-  private claimTransaction = new Mutex(1)
+  // Single queue for all incoming settlements, outgoing settlements and channel claims
+  private queue = new Queue()
   // Timer/interval for channel watcher to claim incoming, settling channels
-  private watcher?: NodeJS.Timer
-
-  constructor (opts: {
+  private watcher: NodeJS.Timer | null
+
+  constructor ({
+    accountName,
+    accountData,
+    master,
+    sendMessage,
+    dataHandler,
+    moneyHandler
+  }: {
     accountName: string,
+    accountData?: AccountData,
     master: EthereumPlugin,
     // Wrap _call/expose method to send WS messages
     sendMessage: (message: BtpPacket) => Promise<BtpPacketData>,
     dataHandler: DataHandler,
     moneyHandler: MoneyHandler
   }) {
-    this.master = opts.master
-    this.sendMessage = opts.sendMessage
-    this.dataHandler = opts.dataHandler
-    this.moneyHandler = opts.moneyHandler
-
-    // No change detection until after `connect` is called
-    this.account = {
-      accountName: opts.accountName,
+    this.master = master
+    this.sendMessage = sendMessage
+    this.dataHandler = dataHandler
+    this.moneyHandler = moneyHandler
+
+    this.account = new Proxy({
+      accountName,
       balance: new BigNumber(0),
       payoutAmount: this.master._balance.settleTo.gt(0)
-        // If we're prefunding, we don't care about total fulfills
+        // If we're prefunding, we don't care about the total amount fulfilled
         // Since we take the min of this and the settleTo/settleThreshold delta, this
         // essentially disregards the payout amount
         ? new BigNumber(Infinity)
         // If we're settling up after fulfills, then we do care
-        : new BigNumber(0)
+        : new BigNumber(0),
+      ...accountData
+    }, {
+      set: (account, key, val) => {
+        // Commit the changes to the configured store
+        this.master._store.set(`${accountName}:account`, {
+          ...account,
+          [key]: val
+        })
+
+        return Reflect.set(account, key, val)
+      }
+    })
+
+    this.watcher = this.startChannelWatcher()
+  }
+
+  private async fetchChannel (channelId: string): Promise<IChannel | undefined> {
```

Comment:
Maybe call this `updateChannelChache` to avoid the double naming and make it clear that this does more than a simple fetch.